core/src/air/public_values.rs

```diff
@@ -27,59 +28,73 @@ pub struct PublicValues<W, T> {
     pub exit_code: T,
 }
 
-impl<F: AbstractField> PublicValues<Word<F>, F> {
-    pub fn new(other: PublicValues<u32, u32>) -> Self {
-        let PublicValues {
-            committed_value_digest,
-            shard,
-            start_pc: first_row_pc,
-            next_pc: last_row_next_pc,
-            exit_code,
-        } = other;
-        Self {
-            committed_value_digest: committed_value_digest.map(Word::from),
-            shard: F::from_canonical_u32(shard),
-            start_pc: F::from_canonical_u32(first_row_pc),
-            next_pc: F::from_canonical_u32(last_row_next_pc),
-            exit_code: F::from_canonical_u32(exit_code),
-        }
-    }
-}
-
-impl<T: Clone + Debug> PublicValues<Word<T>, T> {
-    pub fn to_vec(&self) -> Vec<T> {
-        self.committed_value_digest
+impl PublicValues<u32, u32> {
+    /// Convert the public values into a vector of field elements.  This function will pad the vector
+    /// to the maximum number of public values.
+    pub fn to_vec<F: AbstractField>(&self) -> Vec<F> {
```

Comment:
ya, i think the [F; PROOF_MAX_NUM_PVS] is safer.  There are actually already checks for  in the serialize and deserialize that checks for that size, but ya, having an explicit size is safer.

I actually was trying that initially, but it will require a custom serialize and deserialize serde function.  But that additional code complexity is probably worth it.